File: app/src/main/java/com/theone/music/ui/fragment/signer/SignerSearchFragment.java
package com.theone.music.ui.fragment.signer;//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛

import android.os.Bundle;

import com.theone.common.constant.BundleConstant;
import com.theone.music.data.model.Singer;
import com.theone.music.ui.fragment.BaseMusicFragment;
import com.theone.music.viewmodel.SingerSearchViewModel;

/**
 * @author The one
 * @date 2022-04-11 14:06
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */
public class SignerSearchFragment extends BaseMusicFragment<SingerSearchViewModel> {

    private SignerSearchFragment() {
    }

    public static SignerSearchFragment newInstance(Singer singer) {
        SignerSearchFragment fragment = new SignerSearchFragment();
        Bundle bundle = new Bundle();
        bundle.putParcelable(BundleConstant.DATA, singer);
        fragment.setArguments(bundle);
        return fragment;
    }

    private Singer mSinger;

    @Override
    protected String getTopBarTitle() {
        return mSinger.getName();
    }

    @Override
    public void initData() {
        mSinger = getArguments().getParcelable(BundleConstant.DATA);
        String type;
        String url = mSinger.getUrl();
        type = url.substring(url.indexOf("-") + 1, url.indexOf("."));
        getViewModel().setType(Integer.parseInt(type));
    }

}

File: app/src/main/java/com/theone/music/ui/fragment/signer/SignerSearchFragment.kt
package com.theone.music.ui.fragment.signer

import android.util.Log
import com.theone.common.constant.BundleConstant
import com.theone.common.ext.bundle
import com.theone.common.ext.getNumbers
import com.theone.common.ext.getValueNonNull
import com.theone.common.ext.matchResult
import com.theone.music.data.model.Singer
import com.theone.music.ui.fragment.BaseMusicFragment
import com.theone.music.ui.fragment.rank.RankItemFragment
import com.theone.music.viewmodel.RankViewModel
import com.theone.music.viewmodel.SingerSearchViewModel
import com.theone.mvvm.ext.qmui.setTitleWitchBackBtn

//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛
/**
 * @author The one
 * @date 2022-04-07 09:46
 * @describe 歌
 * @email 625805189@qq.com
 * @remark
 */
class SignerSearchFragment private constructor() :
    BaseMusicFragment<SingerSearchViewModel>() {

    companion object {
        fun newInstance(singer: Singer): SignerSearchFragment = SignerSearchFragment().bundle {
            putParcelable(BundleConstant.DATA, singer)
        }
    }

    private val mSinger: Singer by getValueNonNull(BundleConstant.DATA)

    override fun initData() {
        setTitleWitchBackBtn(mSinger.name)
        val type = mSinger.url.run {
            substring(indexOf("-") + 1, indexOf("."))
        }
        getViewModel().type = type.toInt()
    }
}